src/AdminRoatp/SFA.DAS.AdminRoatp.Api.UnitTests/Controllers/OrganisationsControllerTests.cs

```diff
@@ -26,4 +26,36 @@ public async Task GetOrganisations(
         mediatorMock.Verify(m => m.Send(It.Is<GetOrganisationsQuery>(r => r.SearchTerm == request.SearchTerm), It.IsAny<CancellationToken>()), Times.Once());
         response.Value.Should().BeEquivalentTo(expectedResponse);
     }
-}
+
+    [Test, MoqAutoData]
+    public async Task GetOrganisationReturnSuccessfulResponse(
+        [Frozen] Mock<IMediator> mediatorMock,
+        [Greedy] OrganisationController sut,
+        GetOrganisationQuery request,
+        GetOrganisationQueryResponse expectedResponse)
+    {
+        mediatorMock.Setup(m => m.Send(It.IsAny<GetOrganisationQuery>(), It.IsAny<CancellationToken>())).ReturnsAsync(expectedResponse);
+
+        var result = await sut.GetOrganisation(request.ukprn, It.IsAny<CancellationToken>());
+        var response = (OkObjectResult)result;
+
+        mediatorMock.Verify(m => m.Send(It.Is<GetOrganisationQuery>(r => r.ukprn == request.ukprn), It.IsAny<CancellationToken>()), Times.Once());
+        response.Value.Should().BeEquivalentTo(expectedResponse);
+    }
+
+    [Test, MoqAutoData]
+    public async Task GetOrganisationReturnNotFoundResponse(
+        [Frozen] Mock<IMediator> mediatorMock,
+        [Greedy] OrganisationController sut,
+        GetOrganisationQuery request)
+    {
+        GetOrganisationQueryResponse? expectedResponse = null;
+        mediatorMock.Setup(m => m.Send(It.IsAny<GetOrganisationQuery>(), It.IsAny<CancellationToken>())).ReturnsAsync(expectedResponse);
+
+        var result = await sut.GetOrganisation(request.ukprn, It.IsAny<CancellationToken>());
+        var response = (NotFoundResult)result;
+
+        mediatorMock.Verify(m => m.Send(It.Is<GetOrganisationQuery>(r => r.ukprn == request.ukprn), It.IsAny<CancellationToken>()), Times.Once());
+        response.Should().BeOfType<NotFoundResult>();
```

Comment:
response is already casted to NotFoundResult so this will never fail. Instead delete `response` variable and use `result` here.